From a78f2f0feda8f92cb59afe8236bd90726908768f Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 25 Jun 2017 00:59:24 -0700
Subject: [PATCH] tools.cc: fixed unused-result warning

fix
| ../../squid-3.5.26/src/tools.cc: In function 'void enter_suid()':
| ../../squid-3.5.26/src/tools.cc:616:11: error: ignoring return value of 'int setuid(__uid_t)', declared with attribute warn_unused_result [-Werror=unused-result]
|      setuid(0);
|      ~~~~~~^~~

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
Upstream-Status: Submitted

 src/tools.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/tools.cc b/src/tools.cc
index 8137a03..843e266 100644
--- a/src/tools.cc
+++ b/src/tools.cc
@@ -612,8 +612,8 @@ enter_suid(void)
     if (setresuid((uid_t)-1, 0, (uid_t)-1) < 0)
         debugs (21, 3, "enter_suid: setresuid failed: " << xstrerror ());
 #else
-
-    setuid(0);
+    if (setuid(0) < 0)
+        debugs(50, DBG_IMPORTANT, "WARNING: no_suid: setuid(0): " << xstrerror());
 #endif
 #if HAVE_PRCTL && defined(PR_SET_DUMPABLE)
     /* Set Linux DUMPABLE flag */
-- 
2.13.1

